### PR TITLE
use latest alpine version

### PR DESCRIPTION
**Pull request by julianlg92**
upgrade aws_cli to v2

use multi stage building to reduce size

add helm cli

### CODE REVIEW

**Review by kritchie — APPROVED** [review]
LGTM ! :+1:

**Comment by adrian-chang-alcover** [comment]
I got this error

```
[+] Building 3.0s (22/23)                                                                                                                                       
 => CACHED [internal] load remote build context                                                                                                            0.3s
 => [internal] load metadata for docker.io/library/alpine:3.17                                                                                             0.0s
 => [internal] load metadata for docker.io/library/python:alpine3.17                                                                                       0.0s
 => [builder  1/14] FROM docker.io/library/python:alpine3.17                                                                                               0.0s
 => [main 1/6] FROM docker.io/library/alpine:3.17                                                                                                          0.0s
 => CACHED [builder  2/14] RUN apk add --no-cache git unzip groff build-base libffi-dev cmake                                                              0.0s
 => CACHED [builder  3/14] RUN git clone --single-branch --depth 1 -b 2.9.19 https://github.com/aws/aws-cli.git                                            0.0s
 => CACHED [builder  4/14] WORKDIR aws-cli                                                                                                                 0.0s
 => CACHED [builder  5/14] RUN python -m venv venv                                                                                                         0.0s
 => CACHED [builder  6/14] RUN . venv/bin/activate                                                                                                         0.0s
 => CACHED [builder  7/14] RUN scripts/installers/make-exe                                                                                                 0.0s
 => CACHED [builder  8/14] RUN unzip -q dist/awscli-exe.zip                                                                                                0.0s
 => CACHED [builder  9/14] RUN aws/install --bin-dir /aws-cli-bin                                                                                          0.0s
 => CACHED [builder 10/14] RUN /aws-cli-bin/aws --version                                                                                                  0.0s
 => CACHED [builder 11/14] RUN rm -rf     /usr/local/aws-cli/v2/current/dist/aws_completer     /usr/local/aws-cli/v2/current/dist/awscli/data/ac.index     0.0s
 => CACHED [builder 12/14] RUN find /usr/local/aws-cli/v2/current/dist/awscli/data -name completions-1*.json -delete                                       0.0s
 => CACHED [builder 13/14] RUN find /usr/local/aws-cli/v2/current/dist/awscli/botocore/data -name examples-1.json -delete                                  0.0s
 => CACHED [builder 14/14] RUN wget https://get.helm.sh/helm-v3.11.0-linux-amd64.tar.gz -O - | tar -xz &&     mv linux-amd64/helm /usr/bin/helm &&     ch  0.0s
 => CACHED [main 2/6] COPY --from=builder /usr/local/aws-cli/ /usr/local/aws-cli/                                                                          0.0s
 => CACHED [main 3/6] COPY --from=builder /aws-cli-bin/ /usr/local/bin/                                                                                    0.0s
 => CACHED [main 4/6] COPY --from=builder /usr/bin/helm /usr/bin/                                                                                          0.0s
 => ERROR [main 5/6] RUN apk --update --no-cache add     python     jq     bash     git     curl     nodejs     npm     groff     less     && npm install  2.6s
------
 > [main 5/6] RUN apk --update --no-cache add     python     jq     bash     git     curl     nodejs     npm     groff     less     && npm install -g yarn     && helm plugin install https://github.com/hypnoglow/helm-s3.git:
#0 0.948 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/x86_64/APKINDEX.tar.gz
#0 1.559 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/x86_64/APKINDEX.tar.gz
#0 1.776 ERROR: unable to select packages:
#0 1.868   python (no such package):
#0 1.868     required by: world[python]
------
ERROR: failed to solve: executor failed running [/bin/sh -c apk --update --no-cache add     python     jq     bash     git     curl     nodejs     npm     groff     less     && npm install -g yarn     && helm plugin install https://github.com/hypnoglow/helm-s3.git]: exit code: 1
```

**Comment by adrian-chang-alcover** [comment]
Since there are some Bitbucket Pipelines using the `latest` tag of this image I propose the following:
1. rename current `latest` tag to `1.0.0` and update the current pipelines using the `latest` tag
2. create the new version with tag `2.0.0`
3. create a Changelog.md file specifying the tools (docker, alpine, aws, helm) versions each Image tag contains.

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
+WORKDIR aws-cli
+RUN python -m venv venv
+RUN . venv/bin/activate
+RUN scripts/installers/make-exe
+RUN unzip -q dist/awscli-exe.zip
+RUN aws/install --bin-dir /aws-cli-bin
+RUN /aws-cli-bin/aws --version
+
+# reduce image size: remove autocomplete and examples
+RUN rm -rf \
```
Comment (adrian-chang-alcover) [inline Dockerfile:20]: I think each `RUN` creates an image layer, even if you remove these files they remains on previously layers causing the image size not decreasing.
That's why the remove is done on the same `RUN` step.

**Review by adrian-chang-alcover — APPROVED** [review]

**Pull request merged**